3.2/root/usr/share/container-scripts/mongodb/init-petset-replset.sh

```diff
@@ -84,7 +84,12 @@ function setup_primary() {
     local config="{_id: \"${MONGODB_REPLICA_NAME}\", members: [ {_id: 0, host: \"${member_host}\"} ]}"
 
     info "Initiating MongoDB replica using: ${config}"
-    mongo admin --eval "rs.initiate(${config});${WAIT_PRIMARY_STATUS}" --quiet
+    mongo admin --eval "
+      if (db.isMaster().setName != \"${MONGODB_REPLICA_NAME}\") {
+        rs.initiate(${config});
+        ${WAIT_PRIMARY_STATUS}
+      }
+    " --quiet
```

Comment:
Yeah, I tested by scaling down to 0 then back up to 3. I was a bit in a hurry, so possibly when I scaled back up there was still a pod in termination phase... as I considered before on IRC, there might be different behavior depending on the replica set state during the termination.

I tested again now with both `oc delete pods --all` and `oc scale --replicas=0/3 petset mongodb`, and in both cases it lost data :(